src/Framework/Collection.php

```diff
@@ -0,0 +1,21 @@
+<?php
+namespace Give\Framework;
+
+/**
+ * Class Collection
+ *
+ * Interface that represents a collection.
+ *
+ * @since 2.10.0
+ * @package Give\Framework
+ */
+interface Collection {
```

Comment:
I want to be careful how we use the term "Collection" which, to my understanding, usually communicates a data structure that is traversable and countable, see the [Data Structure Collection interface](https://www.php.net/manual/en/class.ds-collection.php).

Instead, looking at the below implementation, I think this more so falls under the concept of a Provider. Similar to what we are doing in Test Data, each provider returns the data in a known structure.